src/OpenTelemetry.Instrumentation.Wcf/WcfInstrumentationOptions.cs

```diff
@@ -35,23 +35,23 @@ public class WcfInstrumentationOptions
     /// <para>object: the raw <see cref="Message"/> from which additional information can be extracted to enrich the activity.
     /// </para>
     /// </remarks>
-    public Action<Activity, string, object> Enrich { get; set; }
+    public Action<Activity, string, object>? Enrich { get; set; }
 
     /// <summary>
     /// Gets or sets a Filter function to filter instrumentation for requests on a per request basis.
     /// The Filter gets the Message, and should return a boolean.
     /// If Filter returns true, the request is collected.
     /// If Filter returns false or throw exception, the request is filtered out.
     /// </summary>
-    public Func<Message, bool> IncomingRequestFilter { get; set; }
+    public Func<Message, bool>? IncomingRequestFilter { get; set; }
 
     /// <summary>
     /// Gets or sets a Filter function to filter instrumentation for requests on a per request basis.
     /// The Filter gets the Message, and should return a boolean.
     /// If Filter returns true, the request is collected.
     /// If Filter returns false or throw exception, the request is filtered out.
     /// </summary>
-    public Func<Message, bool> OutgoingRequestFilter { get; set; }
+    public Func<Message, bool>? OutgoingRequestFilter { get; set; }
```

Comment:
Other parts of public contract looks good.

The OutgoingRequestFilter seems tricky to me. There is no null check for Message in our code, and based on some short check, there is a chance that this message can be null.

I would consider changing it to, but maybe there is a better way to handle it on our side.

It is related not only to this place, but all places where outgoing requests are handled.

```suggestion
    public Func<Message?, bool>? OutgoingRequestFilter { get; set; }
```

Reply:
IMHO logically it doesn't make sense for a filter to ever get an outgoing request that doesn't exist - there's nothing to filter out. As a general rule I haven't been putting null guards on non-public interfaces which are only called by the WCF framework, but if we're worried about WCF sending us a null outgoing message then I would rather put null-guards on all the entry points than to pollute the public-facing interface....unless of course there's actually a valid use case where an outgoing message could be null that I don't know about?

Reply:
hmmmm actually yeah thinking about this I guess it's possible someone else could inject their own binding element which passes bad params to ours....I will add null guards